Feature/tcgen

OliviaOstwald-NOAA opened the pull request:
## Description of Changes

> Please include a summary of the changes and the related GitHub issue(s). Please also include relevant motivation and context.

- Updating $YYYY to 2024 

**scripts/stats/hurricane** 
- Set INIT_FREQ and VALID_FREQ=6

**scripts/plots/hurricane**
- Updated commands to grab OPS_HIT data from METplus output files instead of DEV_HIT
- Updated file names to "ops" from "dev" where appropriate

**METplus variable changes in TCGen_template.conf:**
- Set forecast hour window end to 168h (from 120h because 5 day genesis outlooks were discontinued in 2023 and replaced with 7 day)
- Set MATCH_POINT_TO_TRACK = true

The purpose of this PR is to update the global_det_tcgen scripts, which will be used during the official GFSv17 evaluation, and for verification analysis during and after the 2025 Hurricane Season.

## Developer Questions and Checklist
* Is this a high priority PR? If so, why and is there a date it needs to be merged by?
* Do you have any planned upcoming annual leave/PTO? 
* Are there any changes needed in the times when the jobs are supposed to run/kick-off?
  
- [ ] The code changes follow [NCO's EE2 Standards](https://www.nco.ncep.noaa.gov/idsb/implementation_standards/ImplementationStandards.v11.0.0.pdf).
- [ ] Developer's name is removed throughout the code and have used `${USER}` where necessary throughout the code.
- [ ] References the feature branch for `HOMEevs` are removed from the code.
- [ ] J-Job environment variables, COMIN and COMOUT directories, and output follow what has been [defined](https://docs.google.com/document/d/1JWg_4q80aYmmAoD21GFjp9R9y5-3w7WGM3-0HJk0Pjs/edit#heading=h.7ysbr191vzu4) for EVS.
- [ ] Jobs over 15 minutes in runtime have restart capability.
- [ ] If applicable, changes in the `dev/drivers/scripts` or `dev/modulefiles` have been made in the corresponding `ecf/scripts` and `ecf/defs/evs-nco.def`? 
- [ ] Jobs contain the appropriate file checking and don't run METplus for any missing data.
- [ ] Code is using METplus wrappers structure and not calling MET executables directly.
- [ ] Log is free of any ERRORs or WARNINGs.

## Testing Instructions

1. Copy branch in local EVS workspace
2. Set $USER to olivia.ostwald OR copy data directory "evs_tc_2024" to personal noscrub space:
/lfs/h2/emc/vpppg/noscrub/olivia.ostwald/evs_tc_2024
3. Run for stats:
**jevs_hurricane_global_det_tcgen_stats.sh**
4. Once stats job is complete, run for plots:
**jevs_hurricane_global_det_tcgen_plots.sh**

> Please include testing instructions for the PR assignee. Include all relevant input datasets needed to run the tests.

OliviaOstwald-NOAA committed:
Added UKMN

OliviaOstwald-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/tcgen

OliviaOstwald-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/tcgen

OliviaOstwald-NOAA committed:
Merge branch 'develop' into feature/tcgen

OliviaOstwald-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/tcgen

OliviaOstwald-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/tcgen

OliviaOstwald-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/tcgen

OliviaOstwald-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/tcgen

OliviaOstwald-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/tcgen

OliviaOstwald-NOAA committed:
Successful TCgen run for 2024!!

OliviaOstwald-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/tcgen

OliviaOstwald-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/tcgen

OliviaOstwald-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/tcgen

OliviaOstwald-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/tcgen

OliviaOstwald-NOAA committed:
TC gen 2022 runs

OliviaOstwald-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/tcgen

OliviaOstwald-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/tcgen

OliviaOstwald-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/tcgen

OliviaOstwald-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/tcgen

OliviaOstwald-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/tcgen

OliviaOstwald-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/tcgen

OliviaOstwald-NOAA committed:
PDY set to 20241231;gen ops window to 120h,match point to true

OliviaOstwald-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/tcgen

OliviaOstwald-NOAA committed:
Testing METplus config settings

OliviaOstwald-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/tcgen

OliviaOstwald-NOAA committed:
Merge branch 'develop' into feature/tcgen

OliviaOstwald-NOAA committed:
Merge branch 'feature/tcgen' of https://github.com/OliviaOstwald-NOAA/EVS into feature/tcgen

OliviaOstwald-NOAA committed:
replace grep FYOY/N lines with awk '$NF == FYOY/N'

OliviaOstwald-NOAA committed:
Changes made for testing, not final version yet

OliviaOstwald-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/tcgen

OliviaOstwald-NOAA committed:
Final set of tests,will decide on FCST_HR_BEG setting

OliviaOstwald-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/tcgen

OliviaOstwald-NOAA committed:
OPS_HIT_WINDOW_END=48;FCST_HR_BEG=6

OliviaOstwald-NOAA committed:
OPS_HIT_END=48,MIN_DUR=24,FCST_HR_BEG=6

OliviaOstwald-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/tcgen

OliviaOstwald-NOAA committed:
FCST_HR_BEG=0,removed PROB_THRESH options

OliviaOstwald-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/tcgen

OliviaOstwald-NOAA committed:
Tests with trimmed genesis file (exclude non-storm days)

OliviaOstwald-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/tcgen

AliciaBentley-NOAA commented:
@OliviaOstwald-NOAA I am starting to work on this PR! I'll send updates as I have them. Thanks!

CC @sbanihash @AndrewBenjamin-NOAA

AliciaBentley-NOAA commented on dev/drivers/scripts/stats/hurricane/jevs_hurricane_global_det_tcgen_stats.sh:
```diff
@@ -33,14 +33,15 @@ module load prod_envir/${prod_envir_ver}
-export COMINbdeckNHC=/lfs/h2/emc/vpppg/noscrub/$USER/evs_tc_2023/bdeck
-export COMINadeckJTWC=/lfs/h2/emc/vpppg/noscrub/$USER/evs_tc_2023/adeck
-export COMINbdeckJTWC=/lfs/h2/emc/vpppg/noscrub/$USER/evs_tc_2023/bdeck
+#export COMINgenesis=/lfs/h2/emc/vpppg/noscrub/$USER/evs_tc_2024/TCgen
+export COMINgenesis=/lfs/h2/emc/vpppg/noscrub/olivia.ostwald/evs_tc_2024/tests/7-day-0
```
@OliviaOstwald-NOAA Do we need to switch this back to be the commented out line?

OliviaOstwald-NOAA replied:
This is from testing I did over the weekend because the PR review process hadn't yet started. I will remove this line and uncomment line 39.

I can provide more information on what I was testing and why, if needed.

AliciaBentley-NOAA commented on parm/metplus_config/stats/hurricane/TCGen_template.conf:
```diff
@@ -121,8 +121,8 @@ TC_GEN_DLAND_THRESH = >0
 
-#TC_GEN_GENESIS_MATCH_POINT_TO_TRACK = True
-TC_GEN_GENESIS_MATCH_POINT_TO_TRACK = false
+TC_GEN_GENESIS_MATCH_POINT_TO_TRACK = True
+#TC_GEN_GENESIS_MATCH_POINT_TO_TRACK = false
```
@OliviaOstwald-NOAA Different settings below this use TRUE and FALSE. Should these lines be capitalized (and does the capitalization make a difference)?

OliviaOstwald-NOAA replied:
This is how it is in the current code package (in develop branch) and was not something I tested. I will test this now to see if capitalization makes a difference and confirm the results below (either yes or no).

OliviaOstwald-NOAA replied:
Tested and it makes no difference. I'm inclined to keep it as it is because there are numerous other settings without capitalization. If you want, I can go ahead and change everything to be capitalized.

sbanihash replied:
@OliviaOstwald-NOAA, unrelated to Alicia's question here, but does this even get used in the OPS_CAT that your files are using?

OliviaOstwald-NOAA replied:
@sbanihash the current version in develop has line 124 commented out so I kept it. It is not needed and can be removed if that is preferred.

AliciaBentley-NOAA replied:
@OliviaOstwald-NOAA Please remove the commented out line. Thanks!

AliciaBentley-NOAA commented on scripts/plots/hurricane/exevs_hurricane_global_det_tcgen_plots.sh:
```diff
@@ -7,8 +7,10 @@ export cartopyDataDir=${cartopyDataDir:-/apps/ops/prod/data/cartopy}
 export TCGENdays="TC Genesis(05/01/${YEAR}-11/30/${YEAR})"
-export basinlist="al ep wp"
-export modellist="gfs ecmwf cmc"
+#export basinlist="al ep wp"
+export basinlist="al"
+#export modellist="gfs ecmwf cmc"
+export modellist="gfs"
```
@OliviaOstwald-NOAA Should we test on all three basins and all three global models?

OliviaOstwald-NOAA replied:
Whoops, those are from my weekend tests (to save time) and I forgot to remove them. I have already tested the package for all three models and basins prior to submitting the draft PR and opening it as "ready for review". I will remove line 11 and 13 and uncomment lines 10 and 12.

AliciaBentley-NOAA commented on scripts/plots/hurricane/exevs_hurricane_global_ens_tropcyc_plots.sh:
```diff
@@ -3,9 +3,11 @@ set -x
-export basinlist="al ep wp"
-export numlist="01 02 03 04 05 06 07 08 09 10 11 12 13 14 15 16 17 18 19 20 \
-	        21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40"
+export basinlist="al ep"
+#export basinlist="al ep wp"
```
@OliviaOstwald-NOAA Similar to above, should be test on all three basins? Minimally, I think this line should be the same as what we have set in `scripts/plots/hurricane/exevs_hurricane_global_det_tcgen_plots.sh`.

OliviaOstwald-NOAA replied:
Same thing as my previous comment, this was just for recent testing and I forgot to remove line 6 and uncomment line 7. I will update this and push those changes.

OliviaOstwald-NOAA replied:
Same as above for the most part, this was accidentally leftover from some test during the past few months or was mistakenly changed. Thank you for catching this, and I will change it back!

AliciaBentley-NOAA commented on scripts/stats/hurricane/exevs_hurricane_global_det_tcgen_stats.sh:
```diff
@@ -4,8 +4,10 @@ export PS4=' + exevs_hurricane_global_det_tcgen_stats.sh line $LINENO: '
 export YEAR=${YYYY}
-export basinlist="al ep wp"
-export modellist="gfs ecmwf cmc"
+#export basinlist="al ep wp"
+export basinlist="al"
+#export modellist="gfs ecmwf cmc"
+export modellist="gfs"
```
@OliviaOstwald-NOAA Should we test on all basins and all three global models?

OliviaOstwald-NOAA replied:
Same as above (accidentally leftover from weekend testing and all basins and models have already been tested with the METplus settings we decided on).

AliciaBentley-NOAA commented on scripts/stats/hurricane/exevs_hurricane_global_ens_spread_stats.sh:
```diff
@@ -6,9 +6,11 @@ export MetOnMachine=${MetOnMachine:-$MET_ROOT}
-export numlist="01 02 03 04 05 06 07 08 09 10 11 12 13 14 15 16 17 18 19 20 \
-	        21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40"
+export basinlist="al ep"
+#export basinlist="al ep wp"
```
@OliviaOstwald-NOAA Should we test on all three basins?

OliviaOstwald-NOAA replied:
Same as above, except no global_ens jobs were tested. This is likely something that was leftover during the few months I was working on this PR. Thanks for catching this! 

The global_ens package has already been tested successfully for all three basins so I assure you there are no issues.

AliciaBentley-NOAA commented on scripts/stats/hurricane/exevs_hurricane_global_ens_tropcyc_stats.sh:
```diff
@@ -6,9 +6,11 @@ export MetOnMachine=${MetOnMachine:-$MET_ROOT}
-export basinlist="al ep wp"
-export numlist="01 02 03 04 05 06 07 08 09 10 11 12 13 14 15 16 17 18 19 20 \
-	        21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40"
+export basinlist="al ep"
+#export basinlist="al ep wp"
```
@OliviaOstwald-NOAA Should we test on all three basins?

OliviaOstwald-NOAA replied:
Same as above, this PR is only for global_det_tcgen and was accidentally not changed back at some point during the past few months. Thank you for catching this!

OliviaOstwald-NOAA committed:
Removed basin,model,data lines used for misc testing

AliciaBentley-NOAA commented:
@OliviaOstwald-NOAA Thank you for submitting your updates! Are we ready to begin testing this PR or are additional changes coming today?

OliviaOstwald-NOAA commented:
@AliciaBentley-NOAA ready to begin testing! Copying the run instructions below: 

1. Copy branch in local EVS workspace
2. Set $USER to olivia.ostwald OR copy data directory "evs_tc_2024" to personal noscrub space:
/lfs/h2/emc/vpppg/noscrub/olivia.ostwald/evs_tc_2024
3. Run for stats:**jevs_hurricane_global_det_tcgen_stats.sh**
4. Once stats job is complete, run for plots:
**jevs_hurricane_global_det_tcgen_plots.sh**

AliciaBentley-NOAA commented:
@OliviaOstwald-NOAA Starting on the code now! Please sync your fork. Thanks!

AliciaBentley-NOAA commented:
@OliviaOstwald-NOAA The hurricane tcgen job is running! You can follow along here. Please check the output when the job completes for ERRORs, WARNINGs, exceeded, etc. Please also compare the final stat files to your final stat files.

### jevs_hurricane_global_det_tcgen_stats.sh
**Log file:** /lfs/h2/emc/vpppg/noscrub/alicia.bentley/pr716test/EVS/dev/drivers/scripts/stats/hurricane/jevs_hurricane_global_det_tcgen_stats.o195868961
**DATA:** /lfs/h2/emc/stmp/alicia.bentley/evs_test/dev/tmp/jevs_hurricane_global_det_tcgen_stats_00.195868961.cbqs01/
**Stats:** /lfs/h2/emc/vpppg/noscrub/alicia.bentley/evs/v2.0/stats/hurricane/

CC @sbanihash

AliciaBentley-NOAA commented:
@OliviaOstwald-NOAA Your hurricane stats job completed. I do not see any WARNINGs or ERRORs. I don't see anything was exceeded. I do think that we can trim down the walltime, though. Your juob ran for 3 minutes and requested 30 minutes. Can your job request 10 minutes? Thank you!

```
grep walltime jevs_hurricane_global_det_tcgen_stats.o195868961
    resources_used.walltime = 00:03:33
    Resource_List.walltime = 00:30:00
```

CC @sbanihash

OliviaOstwald-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/tcgen

OliviaOstwald-NOAA committed:
Updated walltimes

AndrewBenjamin-NOAA commented on scripts/plots/hurricane/exevs_hurricane_global_det_tcgen_plots.sh:
```diff
@@ -41,9 +41,9 @@ if [ ! -d ${OUTPUT} ]; then mkdir -p ${OUTPUT}; fi
 #--- plot the Hits/False Alarms Distribution
 cd ${OUTPUT}
-cp ${USHevs}/hurricane/plots/hits_${basin}.py .
+cp ${USHevs}/${COMPONENT}/hits_${basin}.py .
```
should this be ${COMPONENT}/${STEP}?

OliviaOstwald-NOAA replied:
@AndrewBenjamin-NOAA in develop, ush scripts are under /ush/${COMPONENT} without ${STEP}, same for other components
[https://github.com/OliviaOstwald-NOAA/EVS/tree/develop/ush/hurricane](url)

OliviaOstwald-NOAA commented:
@AliciaBentley-NOAA sorry, just got out of a Hurricane Team meeting! I just synced my feature branch and updated the walltimes for both stats and plots. The plots job takes longer than stats so I changed that walltime to 15 minutes (and stats to 10 minutes). 

Will look at the logfile and stats output from your run now and follow up shortly.

AndrewBenjamin-NOAA commented on scripts/stats/hurricane/exevs_hurricane_global_det_tcgen_stats.sh:
```diff
@@ -52,8 +52,9 @@ fi
 #--- run for TC_gen
 cd ${OUTPUT}
-cp ${PARMevs}/metplus_config/hurricane/stats/TCGen_template.conf .
+cp ${PARMevs}/metplus_config/${STEP}/${COMPONENT}/TCGen_template.conf .
```
Was this meant to be reversed? Should it be 
${COMPONENT}/${STEP}

OliviaOstwald-NOAA replied:
@AndrewBenjamin-NOAA this (/$STEP/$COMPONENT) matches the setup/order in develop

OliviaOstwald-NOAA commented:
@AliciaBentley-NOAA the stats output looks good and matches mine. Before you run plots, there is still a "possible error" in the python plotting scripts that I brought up in our meeting at the end of March:
 
"Possible issue encountered when converting Shape #1541 to GeoJSON: Shapefile format requires that polygons contain at least one exterior ring, but the Shape was entirely made up of interior holes (defined by counter-clockwise orientation in the shapefile format). The rings were still included but were encoded as GeoJSON exterior rings instead of holes."

I misremembered what Saeideh said and thought it was not causing issues. I have looked into this a bit and it's related to how the coordinates are read. The logfile does not have "ERROR" or "WARNING" with this message, but I can look into this further after my V&V presentation this afternoon.

OliviaOstwald-NOAA commented:
@AliciaBentley-NOAA so there is a comment about a known cartopy projection issue in all the python plotting scripts used by **global_det_tcgen_plots** 
[https://github.com/OliviaOstwald-NOAA/EVS/blob/3d814767537d9935f07d8f849cce07b25b4fc407/ush/hurricane/false_al.py#L58](url)

This is not an issue in the other hurricane jobs (because there are no other maps plotted)-I'm not sure if this is something that may have come up in other EVS components? I know it's related to coordinate projection.

OliviaOstwald-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/tcgen

AliciaBentley-NOAA commented:
@OliviaOstwald-NOAA I am going to review the log files from the hurricane stats job this morning and potentially kick off the plots job. I have not investigated the plots error that you are describing. I recommend looking into that error to get to the bottom of it. No other 2D map plots in EVS produce that error, so I think it is something we can fix, although maybe outside of this current PR.

CC @sbanihash

AliciaBentley-NOAA commented:
@OliviaOstwald-NOAA I have looked over the hurricane stats log file. I did not find any instances of the following words: WARNING, ERROR, FAIL, FATAL, killed, exceeded, burst, NOT, cannot, CANNOT, SIGTERM. The job exited cleanly. I also see that the job ran for 3 minutes and 33 seconds, so the new walltime of 10 minutes is consistent with NCO standards. 

One thing I'm not sure about are the contents of the .stat files located in:
/lfs/h2/emc/vpppg/noscrub/alicia.bentley/evs/v2.0/stats/hurricane/global_det/tcgen

The file sizes show that these files can be VERY small (some only have 2 lines). Is this expected?

CC @sbanihash

AliciaBentley-NOAA commented:
@OliviaOstwald-NOAA I have kicked off the plots job to see this message. Note that I pointed to the stat files that I produced in the stats step, not your stat files, to make sure that what we produced in this test is usable in the plots step. 

### jevs_hurricane_global_det_tcgen_plots.sh 
**Log file:** /lfs/h2/emc/vpppg/noscrub/alicia.bentley/pr716test/EVS/dev/drivers/scripts/plots/hurricane/jevs_hurricane_global_det_tcgen_plots.o195949070
**DATA:** /lfs/h2/emc/stmp/alicia.bentley/evs_test/dev/tmp/jevs_hurricane_global_det_tcgen_plots_00.195949070.cbqs01/
**Plots:** /lfs/h2/emc/vpppg/noscrub/alicia.bentley/evs/v2.0/plots/hurricane/global_det/tcgen/

CC @sbanihash @AndrewBenjamin-NOAA @ShannonShields-NOAA

OliviaOstwald-NOAA commented:
@AliciaBentley-NOAA responding about the stats file in this comment (will comment separately about the plots job after checking it!)
The very small stats files are expected-those are used as the input to produce the performance diagrams. There are individual files for each model in each basin, which is why there are so many files, including the very small txt files for cts and ctc output.

OliviaOstwald-NOAA commented:
@AliciaBentley-NOAA I started going through the logfile and the walltime was exceeded while generating the West Pac plots. I don't see any instances of the words you listed in the previous comment, except killed because of the walltime. 

This was run with a walltime of 15 minutes, so I will change it to 25 minutes (to err on the side of caution) and will check the next run to see if it can be reduced to 20 minutes. Let me know if there are any issues with making that change!

OliviaOstwald-NOAA committed:
Increased plots walltime to 25 min

AliciaBentley-NOAA commented:
@OliviaOstwald-NOAA I deleted what was produced by the job that hit its wallclock, pulled in the changes, and started the plot jobs again. 

### jevs_hurricane_global_det_tcgen_plots.sh
**Log file:** /lfs/h2/emc/vpppg/noscrub/alicia.bentley/pr716test/EVS/dev/drivers/scripts/plots/hurricane/jevs_hurricane_global_det_tcgen_plots.o195953691
**DATA:** /lfs/h2/emc/stmp/alicia.bentley/evs_test/dev/tmp/jevs_hurricane_global_det_tcgen_plots_00.195953691.cbqs01/
**Plots:** /lfs/h2/emc/vpppg/noscrub/alicia.bentley/evs/v2.0/plots/hurricane/global_det/tcgen/

CC @sbanihash @AndrewBenjamin-NOAA @ShannonShields-NOAA

OliviaOstwald-NOAA commented:
@AliciaBentley-NOAA checked the logfile and the plots and everything looks good and matches what I generated! 

The logfile has no instances of the words "WARNING, ERROR, FAIL, FATAL, killed, exceeded, burst, NOT, cannot, CANNOT, SIGTERM". The walltime was **18:08** so I will change it to 20 minutes and pull in the most recent commit. 

A comment on the GeoJSON message in the logfile:
**Line 517** is the first instance of the message "Possible error encountered when converting Shape #1541 to GeoJSON..." but there are no ERRORs, WARNINGs, etc. 

As previously mentioned, the ush py plotting scripts used in this job all include comments about a workaround for a known issue with cartopy 
([https://github.com/NOAA-EMC/EVS/blob/6f4a6fba650c70fe35bde7cd84f0970a1de8f574/ush/hurricane/false_al.py#L58](url) ). 
As you mentioned, this doesn't happen in other EVS scripts that generate 2D plots. I would suggest adding this to Fixes and Additions and I can look at other EVS component ush scripts to see how they have coordinates coded and can be addressed in a future PR.

OliviaOstwald-NOAA committed:
Decreased plots walltime to 20 minutes after testing

AliciaBentley-NOAA commented:
@OliviaOstwald-NOAA Please see this list from NCO about a job's runtime and it's requested walltime. An 18 minute job should have a 30 minute walltime. Please update this job to 30 minutes.

Runs for 0-2 minutes = minimum 10 minute wall clock
Runs for 3-10 minutes = minimum 15 minute wall clock
Runs for 11-15 minutes = minimum 25 minute wall clock
Runs for 16-20 minutes = minimum 30 minute wall clock
Runs for 21+ minutes = minimum is (runtime * 1.25) rounded up to the nearest 5 min

OliviaOstwald-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/tcgen

OliviaOstwald-NOAA commented:
Oops, I pushed 20 minutes but will change it to 30 minutes now!

OliviaOstwald-NOAA committed:
Plots walltime set to 30 min per NCO guidelines

AliciaBentley-NOAA commented:
@OliviaOstwald-NOAA Thanks for making that change! Can you please add the "al" hits and false alarms images for gfs, ecmwf, etc., to this PR so that we can see them?

OliviaOstwald-NOAA commented:
@AliciaBentley-NOAA sure thing, just want to make sure I understand. Do you want me to attach the AL hits and false alarms plots for the three models here in the comments?

OliviaOstwald-NOAA committed:
Removed #TC_GEN_GENESIS_MATCH_POINT_TO_TRACK = false

AliciaBentley-NOAA commented:
@OliviaOstwald-NOAA Yes, please add examples of your tcgen plots to this PR for us to save after the directories are deleted. Thanks!

OliviaOstwald-NOAA commented:
@AliciaBentley-NOAA I uploaded all my plots to RZDM in case we want to reference the full set: 

[https://www.emc.ncep.noaa.gov/users/Olivia.Ostwald/2024/tcgen/PR-plots/](url) (clicking the link doesn't work, but you can copy and paste the URL to see the plots!)

**CMC**
![evs hurricane_global_det hitfals al 2024 cmc season tcgen](https://github.com/user-attachments/assets/05c355a5-a9fd-46f2-9ccc-2db0cba775f5)
**ECMWF**
![evs hurricane_global_det hitfals al 2024 ecmwf season tcgen](https://github.com/user-attachments/assets/cd7c8064-efe3-4b0d-b0e2-00b009d7d60a)
**GFS**
![evs hurricane_global_det hitfals al 2024 gfs season tcgen](https://github.com/user-attachments/assets/b51a3a56-1b92-4ebc-9c05-1e5f63bfe298)

AliciaBentley-NOAA commented:
@OliviaOstwald-NOAA Thanks! Can you please add the name of the model (e.g., gfs) be included in the title somewhere? This can be done by adding the $model variable to the line that specifies the image's title. Perhaps tcgen plot titles could look like:

``GFS TC Genesis (05/01/2024-11/30/2024)``

CC @sbanihash @AndrewBenjamin-NOAA

OliviaOstwald-NOAA commented:
@AliciaBentley-NOAA sure thing! I will test that now and reply here with those updated plots.

ShannonShields-NOAA commented:
I agree about the model names in the plot titles as when I initially viewed the graphics, I was confused about what made them different.

OliviaOstwald-NOAA commented:
@AliciaBentley-NOAA @ShannonShields-NOAA agreed, thanks for mentioning this! This is how it was previously coded, but agree this is a good change to make. 

I just did an initial test for just AL GFS that was successful (see plot below). **However,** this fix gets used in the performance diagram (PD) plot as well. So, I will make a new variable for PD titles and test. I will also run the package with all basins and models after to confirm these changes work and all plots are updated before pushing the changes. 

Once everything looks good, I'll comment here letting you know everything is good from my end regarding this fix.

OliviaOstwald-NOAA commented:
@AliciaBentley-NOAA just want to provide an update, I am going to have to create a loop for the plot titles and will likely be done with testing later in the afternoon 

(Thursday, May 15th morning: still working on this, the loops have to be added to the ush scripts and there are 10 python plotting scripts used in this package)

OliviaOstwald-NOAA commented:
@AliciaBentley-NOAA @ShannonShields-NOAA wanted to provide another update, I am still working on adding the model names to the hits/FAs plot title and the basin name to the performance diagram plot title. I've been getting errors where the new variable "is not defined" and am working towards a solution. Just a heads up!

OliviaOstwald-NOAA committed:
Added model names to hits/FAs, basin name to PD plots

OliviaOstwald-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/tcgen

OliviaOstwald-NOAA commented:
@AliciaBentley-NOAA @ShannonShields-NOAA I was able to add the model names to the hits/false alarms plots (as well as just hits and just FAs). I also added the basin name to the title on the performance diagram because they all had the same title ("TC Genesis (5/01/2024-11/30/2024). I am attaching 6 plots here, and have transferred the full set to RZDM (copy and paste URL in browser):
[https://www.emc.ncep.noaa.gov/users/Olivia.Ostwald/2024/tcgen/PR-05152025/](url)

![evs hurricane_global_det hitfals wp 2024 cmc season tcgen](https://github.com/user-attachments/assets/9b8aff01-224a-4541-b70f-4ed98ada2329)
![evs hurricane_global_det hitfals ep 2024 ecmwf season tcgen](https://github.com/user-attachments/assets/d5bd097d-ad9e-40bc-8c57-75d3dd623923)
![evs hurricane_global_det hitfals al 2024 gfs season tcgen (1)](https://github.com/user-attachments/assets/ae266ce7-f358-444f-be5d-1c0bd22189ce)
![evs hurricane_global_det performancediagram wp 2024 season tcgen](https://github.com/user-attachments/assets/ef82e51d-7533-4377-9d23-2b8360d717f1)
![evs hurricane_global_det performancediagram ep 2024 season tcgen](https://github.com/user-attachments/assets/324ff7ae-4a3c-43e3-88ee-6ab99bb98c51)
![evs hurricane_global_det performancediagram al 2024 season tcgen](https://github.com/user-attachments/assets/31253092-cd46-4e4e-8e06-efcde0eb7238)

AliciaBentley-NOAA commented:
Awesome, @OliviaOstwald-NOAA! I will pull in your updates and test the TCGEN plots job again before we wrap this up. We can address the Cartopy message in a later PR. Thanks!

AliciaBentley-NOAA commented:
@OliviaOstwald-NOAA I deleted deleted the plots from my first test, pulled in your changes, and started the plot jobs again.

### jevs_hurricane_global_det_tcgen_plots.sh
**Log file:** /lfs/h2/emc/vpppg/noscrub/alicia.bentley/pr716test/EVS/dev/drivers/scripts/plots/hurricane/jevs_hurricane_global_det_tcgen_plots.o196159426
**DATA:** /lfs/h2/emc/stmp/alicia.bentley/evs_test/dev/tmp/jevs_hurricane_global_det_tcgen_plots_00.196159426.cbqs01/
**Plots:** /lfs/h2/emc/vpppg/noscrub/alicia.bentley/evs/v2.0/plots/hurricane/global_det/tcgen/

CC @sbanihash @AndrewBenjamin-NOAA @ShannonShields-NOAA

OliviaOstwald-NOAA commented:
@AliciaBentley-NOAA everything looks good! The plots match what I generated and all have the model/basin added to the titles (model for hits/FAs, basins for PDs). I see no instances of these words in the logfile: WARNING, ERROR, FAIL, FATAL, killed, exceeded, burst, NOT, cannot, CANNOT, SIGTERM.

OliviaOstwald-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/tcgen

AliciaBentley-NOAA commented:
@OliviaOstwald-NOAA Thank you! I'm going to do a quick check of the log files myself and then approve this PR! 

@ShannonShields-NOAA Can you please be the second reviewer? Andrew requested that we don't make any EVS requests of him and Marcel right now. Thanks!

CC @sbanihash

ShannonShields-NOAA reviewed (approved):
@AliciaBentley-NOAA and @OliviaOstwald-NOAA I have reviewed the TC script changes in this PR and approve for merging. We can now see the basin and model names in the plot titles.

AliciaBentley-NOAA reviewed (approved):
I have commented on the coding changes made in this PR and requested additional updates (which have all been addressed). I also commented on the plots produced. I also examined the stats/plots log files. I approve this PR to be merged. 

CC @OliviaOstwald-NOAA @ShannonShields-NOAA @sbanihash

AliciaBentley-NOAA commented:
@OliviaOstwald-NOAA Please confirm that your fork is synced, and then we can merge this PR! Thanks!

OliviaOstwald-NOAA commented:
@AliciaBentley-NOAA @ShannonShields-NOAA fork is synced!
